src/components/InternalTransfer/index.tsx

```diff
@@ -0,0 +1,184 @@
+import classnames from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Button } from 'react-bootstrap';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { DropdownComponent, Decimal } from '../';
+import { isUsernameEnabled } from '../../api';
+import { CloseIcon } from '../../assets/images/CloseIcon';
+import { Modal } from '../../components';
+import { createInternalTransfersFetch, selectUserInfo, selectWallets, walletsFetch } from '../../modules';
+import { InternalTransferInput } from './InternalInput';
+
+export const InternalTransferComponent = () => {
+    const { formatMessage } = useIntl();
+
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const wallets = useSelector(selectWallets);
+    const user = useSelector(selectUserInfo);
+
+    const [username, setUsername] = useState('');
+    const [currency, setCurrency] = useState('');
+    const [amount, setAmount] = useState('');
+    const [otp, setOtp] = useState('');
+
+    const [show, setShow] = useState(false);
+
+    const BALANCE_FETCH_TIMEOUT = 2000;
+
+    React.useEffect(() => {
+        if (!wallets.length) {
+            dispatch(walletsFetch());
+        }
+    }, [dispatch, wallets]);
+
+    const walletsList = wallets.length ? wallets.map(item => item.currency && item.currency.toUpperCase()) : [];
+
+    const wallet = wallets.length && wallets.find(item => item.currency.toLowerCase() === currency.toLowerCase());
+
+    const balanceError = useMemo(() => classnames('cr-internal-transfer__group--balance', {
+        'cr-internal-transfer__group--error': wallet && wallet.balance && +wallet.balance < +amount,
+    }), [amount, wallet]);
+
+    const translationUsername = isUsernameEnabled() ? 'username' : 'uid';
+
+    const handleCreateTransfer = useCallback(() => {
+        const payload = {
+            currency: currency.toLowerCase(),
+            username_or_uid: username,
+            amount,
+            otp
+        };
+
+        dispatch(createInternalTransfersFetch(payload));
+        setShow(false);
+        setUsername('');
+        setCurrency('');
+        setAmount('');
```

Comment:
1. reset these fields only after successful fetch.
2. You defined function `handleChangeShow()`. Use it please

Reply:
fixed